src/settings/PerAccountNotificationSettingsGroup.js

```diff
@@ -111,9 +128,66 @@ export default function PerAccountNotificationSettingsGroup(props: Props): Node
               },
               subtitle: 'Notifications for this account may not arrive.',
             })()}
-        title="Troubleshooting"
+        title={troubleshootingPageTitle}
         onPress={handleTroubleshootingPress}
       />,
+      <TextRow
+        key="test"
+        title="Send a test notification"
+        onPress={async () => {
+            invariant(pushToken != null, 'should be disabled if pushToken missing');
+            setTestNotificationRequestInProgress(true);
+            try {
+              await api.sendTestNotification(auth, { token: pushToken });
+            } catch (errorIllTyped) {
+              const error: mixed = errorIllTyped; // https://github.com/facebook/flow/issues/2470
+
+              if (!(error instanceof Error)) {
+                logging.error('Unexpected non-error thrown from api.sendTestNotification');
+                return;
+              }
+
+              let msg = undefined;
+              if (error instanceof ApiError) {
+                // TODO recognize error.code and try to fix? give better feedback?
+                msg = _('The server said:\n\n{errorMessage}', { errorMessage: error.message });
+              } else if (error.message.length > 0) {
+                msg = error.message;
+              }
+              showErrorAlert(_('Failed to send test notification'), msg);
+            } finally {
+              setTestNotificationRequestInProgress(false);
+            }
+          }}
+        disabled={(() => {
+            if (zulipFeatureLevel < 217) {
+              return {
+                title: 'Server upgrade required',
+                message: {
+                  text: 'This feature requires {zulipServerVersion} or higher.',
+                  values: { zulipServerVersion: 'Zulip Server 8' },
+                },
+                learnMoreButton: {
+                  url: new URL('https://zulip.readthedocs.io/en/latest/production/upgrade.html'),
+                  text: 'How to upgrade (for server administrators)',
+                },
+              };
+            } else if (pushToken == null) {
+              return {
+                title: 'Cannot send test notification',
+                message: {
+                  text: 'Please tap “{troubleshootingPageTitle}” for more details.',
+                  values: { troubleshootingPageTitle: _(troubleshootingPageTitle) },
+                },
+              };
+            } else if (testNotificationRequestInProgress) {
+              return {
+                title: 'Request in progress',
+                message: 'Please wait. A request is already in progress.',
+              };
+            }
```

Comment:
```suggestion
            } else {
              return false;
            }
```

Or `return undefined`. But either way, when some exit paths from a function return an explicit value (not just `return;`), it's helpful for all of them to do so. That way it's clear that the control path that's returning undefined is intentional.

Reply:
Makes sense.

Actually, when going to make this change, I remembered that you had mentioned on a call a preference for taking this part (and the press handler) out of the JSX, so we can just say:

```jsx
      <TextRow
        key="test"
        title="Send a test notification"
        onPress={handleTestNotificationPress}
        disabled={testNotificationDisabled}
      />,
```

Then `testNotificationDisabled` becomes this:

```js
  let testNotificationDisabled = false;
  if (zulipFeatureLevel < 217) {
    testNotificationDisabled = {
      title: 'Server upgrade required',
      message: {
        text: 'This feature requires {zulipServerVersion} or higher.',
        values: { zulipServerVersion: 'Zulip Server 8' },
      },
      learnMoreButton: {
        url: new URL('https://zulip.readthedocs.io/en/latest/production/upgrade.html'),
        text: 'How to upgrade (for server administrators)',
      },
    };
  } else if (pushToken == null) {
    testNotificationDisabled = {
      title: 'Cannot send test notification',
      message: {
        text: 'Please tap “{troubleshootingPageTitle}” for more details.',
        values: { troubleshootingPageTitle: _(troubleshootingPageTitle) },
      },
    };
  } else if (testNotificationRequestInProgress) {
    testNotificationDisabled = {
      title: 'Request in progress',
      message: 'Please wait. A request is already in progress.',
    };
  }
```

which I think removes the issue with exit paths from a function, since the relevant function (the IIFE) doesn't exist anymore.